soc/xtensa/intel_adsp/ace/power.c

```diff
@@ -173,7 +174,6 @@ void power_gate_entry(uint32_t core_id)
 	soc_cpus_active[core_id] = false;
 	sys_cache_data_flush_range(soc_cpus_active, sizeof(soc_cpus_active));
 	k_cpu_idle();
-	z_xt_ints_off(0xffffffff);
```

Comment:
It's true that after power gating, we are not going back here, but we can't guarantee that power gating will happen. If it's not causing any problems, I would recommend keeping it.

Reply:
The idea of disabling interrupts as a step after WAITI is taken from the reference code. There is no detailed explanation of what exactly can prevent power gating in that case. I think the apprehension about this arises from the fact that firmware does not have full control over all hardware prevents. For example, the HOST can prevent the DSP HP domain from its side without any interaction with firmware.